File: source-connector/src/main/java/sourceconnector/domain/file/S3FileKey.java
package sourceconnector.domain.file;

import lombok.AccessLevel;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
public class S3FileKey implements FileKey {
  private final S3Uri s3Uri;

  @Override
  public String get() {
    return this.s3Uri.toString();
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof S3FileKey)) return false;

    return this.get().equals(((S3FileKey) o).get());
  }

  @Override
  public int hashCode() {
    return this.get().hashCode();
  }
}

File: offset-manager-api/src/main/java/offsetmanager/domain/file/S3FileKey.kt
package offsetmanager.domain.file

class S3FileKey(
  private val s3Uri: S3Uri
) : FileKey {

  override fun get(): String {
    return this.s3Uri.toString()
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other !is S3FileKey) return false

    return this.get() == other.get()
  }

  override fun hashCode(): Int {
    return this.get().hashCode()
  }
}
